app/models/item.rb

```diff
@@ -62,4 +62,24 @@ def return_stock(amount)
   def sell_stock(amount)
     self.inventory -= amount
   end
+
+  def find_discounts
+   merchant.discounts
+  end
+
+  def no_discounts?(quantity)
+    merchant.discounts.where("#{quantity} >= discounts.item_amount") == []
+  end
+
+  def select_highest_discount(quantity)
```

Comment:
this method prioritizes the highest discount by taking the highest percentage of the discounts that items quantity has fulfilled